### PR TITLE
CHK-464: Emit orderForm messages using toast component

### PR DESCRIPTION
**Pull request by lucasecdb**
#### What problem is this solving?

Emit the orderForm messages using Styleguide's toast component, so the user can be aware of the orderForm messages in any page of the store instead of only in `/cart`.

#### How should this be manually tested?

[Workspace](https://toasts--checkoutio.myvtex.com).

To see the toast, add the first item in the shelf and update its quantity to 51.

#### Checklist/Reminders

- [ ] Updated `README.md`.
- [x] Updated `CHANGELOG.md`.
- [x] Linked this PR to a Jira story (if applicable).
- [ ] Updated/created tests (important for bug fixes).
- [ ] Deleted the workspace after merging this PR (if applicable).

#### Type of changes

✔️ | Type of Change
---|---
✔️ | Bug fix 
_ | New feature 
_ | Breaking change 
_ | Technical improvements

### CODE REVIEW

**Comment by darlenedms** [comment]
@lucasecdb is this test failure expected?

**Comment by lucasecdb** [comment]
> @lucasecdb is this test failure expected?

it is not, I need to add a mock module for styleguide

**Comment by wdsrocha** [comment]
Just for the record, the first time I tried to test it I got some bad request errors from graphql

image

Other attempts worked fine, so I'm not sure if this has to do with your PR specifically

File: react/typings/vtex.styleguide.d.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import React from 'react'
+
+declare module 'vtex.styleguide' {
+  export const ToastContext: React.Context<{
+    showToast: (message: string) => void
+    toastState: { isToastVisible: boolean }
+  }>
+}
```
Comment (wdsrocha) [inline react/typings/vtex.styleguide.d.ts:8]: question (non-blocking): just out of curiosity, how hard is to implement this toast typing on styleguide?

Comment (lucasecdb) [reply react/typings/vtex.styleguide.d.ts:8]: I'd need to rewrite the component in typescript

**Review by wdsrocha — APPROVED** [review]

**Review by jeffersontuc — APPROVED** [review]
Nice job 🎉

**Pull request merged**